libseccomp/tests/global_reset.rs

```diff
@@ -0,0 +1,10 @@
+use libseccomp::*;
+
+#[test]
+fn test_reset_global_state() {
+    if check_version(ScmpVersion::from((2, 5, 1))).unwrap() {
+        assert!(reset_global_state().is_ok());
+    } else {
+        assert!(reset_global_state().is_err());
+    }
+}
```

Comment:
Do you have any reason not to put this test in `test/tests.rs`?

Reply:
Yes, because it resets global state, when it runs at the same time as another test in the same process, it will reset the state of that filter and cause the other test to fail.

Is that not why the current makefile uses `--test-threads=1`?

Reply:
I observed the test sometimes failed due to a race condition when I added the userspace notification, so I added `--test-threads=1`  by [this commit](https://github.com/libseccomp-rs/libseccomp-rs/pull/45/commits/83f68755fed05eab2b047a17a6d13167c77df936). However, the frequency of failures is definitely less than at that time, and failures are rarely observed now. I'm not sure the root cause of the failure.

Reply:
On my machine I very consistently see the notify test fail on the first time I run `cargo test` after cleaning, and in particular the global reset test is typically listed within a few tests of it in the log.

This has never occurred for me when they're in separate processes, but does occur when they're in the same process (i.e. when I move the global reset test into `tests/notify.rs`).

In order to make the tests more robust against further additions, I think it makes sense to have the global reset test in its own process, even if we keep --test-threads=1. What do you think?

Reply:
>On my machine I very consistently see the notify test fail on the first time I run cargo test after cleaning, and in particular the global reset test is typically listed within a few tests of it in the log.

Yes, the test will fail without `--test-threads=1`. As I mentioned earlier, before adding `reset_global_state` function, I observed the notify test sometimes failed without `--test-threads=1`. Hence, `--test-threads=1` was a option for the notify APIs tests at that time, not for `reset_global_state`. (Of course, `--test-threads=1` also works for `reset_global_state` test currently.

> In order to make the tests more robust against further additions, I think it makes sense to have the global reset test in its own process, even if we keep --test-threads=1. What do you think?

Yes I agree with you, we should keep `--test-threads=1` to avoid the unexpected failure for notify APIs, even if the global reset test in its own process:)

Reply:
Sorry, I wasn't clear. After moving the notify test to a separate file (which runs as a separate process), I don't experience failures anymore when running `cargo test` without `--test-threads=1`. When I then move the reset global function into the same file as the notify test, the failures reoccur.

Are you able to reproduce the notify test failing even when it's in a separate file?

Reply:
> When I then move the reset global function into the same file as the notify test, the failures reoccur.

Oh, when I move the reset test into same file as notify test and run the test  without --test-threads=1, i cannot observer the failure.
Could you give me your libseccomp library and kernel version?

Reply:
Oh interesting, I don't see it either anymore. Either I was mistaken originally and had moved one of the tests into a different file, or perhaps a recent kernel upgrade. For reference, I'm now on kernel 5.16.10 and libseccomp 2.5.3

Either way, I think it's fine to leave the tests as they are now and keep --test-threads=1.

This completes all the comments for this PR now, I think.